### ISSUE TITLE
Problem with multi targeted dependency with code analysers 

### ISSUE THREAD

**Issue by mattbrailsford** [dotnet/core#6500]
### Description
I seem to be having problems with a multi targeted project which has a dependency on Lucene.net. 

In the .NET Framework target it has a v3.x dependency on the library where as in the .NET Core target it has a v4.x dependency. 

The issue is, the v4.x dependency contains code analysers within it but the v3.x version doesn’t. It seems however if one version has them it expects the other also to have them and so the build errors. 

Metadata file 'C:\Users\...\.nuget\packages\lucene.net\3.0.3\analyzers\dotnet\cs\Lucene.Net.CodeAnalysis.CSharp.dll' could not be found

The only way I’ve found a way round this is to copy the v4 analysers into the v3 package folder and then it works. 

To reproduce you can create a multi targeted project for net472 and net5.0 and then add lucene dependencies to both with net472 having a v3 dependency and net5.0 having a v4 lucene dependency. Attempting to build the project will fail. 

Running in Windows 10 x64, VS 2019

**Comment by mattbrailsford** [comment]
In case it matters, I will point out that in my actual project, Lucene is a transitive dependency. My project has a dependency on Umbraco CMS (v8 for net framework and v9 for net core) which it in turn depends on Lucene.NET.

**Comment by safern** [comment]
Hello @mattbrailsford. Thanks for reporting the issue, this seems like a duplicate of: https://github.com/dotnet/sdk/issues/18148. 

This issue was fixed on: https://github.com/dotnet/sdk/pull/18877 -- and the fix should be available with any .NET 6 Preview 7 SDK, the official Preview7 SDK will come out in a few weeks, however you can install one of our daily builds yourself if you would like to be unblocked. You can do it from here: https://github.com/dotnet/installer/blob/main/README.md#installers-and-binaries and download the one on the `Preview 7` column.

Please let me know if you have any questions. I will close this issue on the meantime.

**Comment by mattbrailsford** [comment]
Ahh, fantastic!

Thank you for the clarification that it’s already been resolved. That’s great news. 🙌🏻
